tests/scripts/all.sh

```diff
@@ -3824,6 +3837,32 @@ component_test_psa_crypto_config_reference_cipher_aead () {
     tests/compat.sh -V NO -p mbedTLS
 }
 
+component_test_full_block_cipher_psa_dispatch () {
+    msg "build: full + PSA dispatch in block_cipher"
+
+    loc_accel_list="ALG_ECB_NO_PADDING \
+                    KEY_TYPE_AES KEY_TYPE_ARIA KEY_TYPE_CAMELLIA"
+
+    # Configure
+    # ---------
+
+    # Start from the full config
+    helper_libtestdriver1_adjust_config "full"
```

Comment:
Hmm, it seems to me we also want to disable `AES_C`, `ARIA_C` and `CAMELLIA_C` (and test that `aes.o` etc are empty after building). After all the goal is to check that these can be completely replaced with drivers and at least a "small" config (that is, one similar to TF-M's medium profile + X.509 and TLS) should still work.

So, it's OK if we have to disable a few things like PKCS5, non-AEAD modes, etc. But we should have AES provided only by a driver and GCM+CCM still working.

As a consequence of dependencies, we might need to then disable a bit more here, and perhaps in the reference component too.

Reply:
I was going for keeping as many things enabled as possible, but you are right: in this was there is no "secure" way to test if block_cipher is dispatching properly or not. I will fix that

Reply:
The main problem in doing this is that legacy modules get automatically enabled in `config_adjust_legacy_from_psa.h`. For example for AES we have:
```
#if defined(PSA_WANT_KEY_TYPE_AES)
#if !defined(MBEDTLS_PSA_ACCEL_KEY_TYPE_AES)
#define PSA_HAVE_SOFT_KEY_TYPE_AES 1
#endif /* !MBEDTLS_PSA_ACCEL_KEY_TYPE_AES */
#if defined(PSA_HAVE_SOFT_KEY_TYPE_AES) || \
    defined(PSA_HAVE_SOFT_BLOCK_MODE) || \
    defined(PSA_HAVE_SOFT_BLOCK_AEAD)
#define MBEDTLS_PSA_BUILTIN_KEY_TYPE_AES 1
#define MBEDTLS_AES_C
#endif /* PSA_HAVE_SOFT_KEY_TYPE_AES || PSA_HAVE_SOFT_BLOCK_MODE */
#endif /* PSA_WANT_KEY_TYPE_AES */
```
We are accelerating AES, but not AEADs nor ciphers (not all of them, only ECB), so we get `AES_C` enabled back.
The only solution I saw for this was to accelerate both AEADs+ciphers, but then we might have TLS using PSA driver instead of BLOCK_CIPHER and this is not what we want for this test :/

Reply:
The definition of G5 for this EPIC is: as a user, I have a driver that accelerates AES (Aria, Camellia) but only the primitive (that is, ECB), not any other mode; I want to be able to remove the built-in implementation of AES (Aria, Camellia) and still get enough things working to support TLS with GCM and CCM ciphersuites.

So, this `all.sh` component should:
- prove that GCM and CCM (built-in) indeed work with a driver that only accelerates ECB and built-in AES (Aria, Camellia) removed;
- clarify what else needs to be disabled / stops working when removing the built-in implementation.